test/check-machines-consoles

```diff
@@ -589,6 +589,98 @@ fullscreen=0
         b.wait_text("#vm-console-vnc-scaling", "Remote resizing")
         wait_widths(lambda remote, local, ui: remote == ui and local == ui)
 
+    def testNonSharedExternal(self):
+        b = self.browser
+        m = self.machine
+
+        # Upload the headless VNC client script to the test machine
+        m.upload(["files/headless-vnc-client.py"], "/tmp/headless-vnc-client.py")
```

Comment:
This feeels like utter overkill. We already have a VNC client -- it's called "cockpit-machines". Can't this open a second browser window with an independent cockpit session (that's in fact `Browser`'s default mode of operation) to simulate concurrent VNC sessions?

Reply:
Cockpit always makes shared connections, we need a non-shared one.

Reply:
It should, actually. We can make a "test-only" secret setting that switches our viewer into non-shared mode. Or maybe we make the detached console always non-shared?

Reply:
I don't think we should open the can of UI worms and expose shared/non-shared to the user.  But "Remote resizing" kinda yells "non-shared", no?

Reply:
Right, see #2268 . That makes headless-vnc-client.py obsolete (we can add a test-only/secret way to make it non-shared, too), so you can ignore all the threads for that.